scripts/ci-entrypoint.sh

```diff
@@ -132,91 +132,104 @@ create_cluster() {
+            fi
+            echo "CCM cluster CIDR: ${CCM_CLUSTER_CIDR:-}"
+
+            export CCM_LOG_VERBOSITY="${CCM_LOG_VERBOSITY:-4}"
+            echo "Installing cloud-provider-azure components via helm"
+            "${HELM}" upgrade cloud-provider-azure --install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure \
+                --set infra.clusterName="${CLUSTER_NAME}" \
+                --set cloudControllerManager.imageRepository="${IMAGE_REGISTRY}" \
+                --set cloudNodeManager.imageRepository="${IMAGE_REGISTRY}" \
+                --set cloudControllerManager.imageName="${CCM_IMAGE_NAME}" \
+                --set cloudNodeManager.imageName="${CNM_IMAGE_NAME}" \
+                --set-string cloudControllerManager.imageTag="${IMAGE_TAG}" \
+                --set-string cloudNodeManager.imageTag="${IMAGE_TAG}" \
+                --set cloudControllerManager.replicas="${CCM_COUNT}" \
+                --set cloudControllerManager.enableDynamicReloading="${ENABLE_DYNAMIC_RELOADING}"  \
+                --set cloudControllerManager.cloudConfig="${CLOUD_CONFIG}" \
+                --set cloudControllerManager.cloudConfigSecretName="${CONFIG_SECRET_NAME}" \
+                --set cloudControllerManager.logVerbosity="${CCM_LOG_VERBOSITY}" \
+                --set-string cloudControllerManager.clusterCIDR="${CCM_CLUSTER_CIDR}" || continue
         fi
-        echo "CCM cluster CIDR: ${CCM_CLUSTER_CIDR:-}"
-
-        export CCM_LOG_VERBOSITY="${CCM_LOG_VERBOSITY:-4}"
-        echo "Installing cloud-provider-azure components via helm"
-        "${HELM}" install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure --generate-name \
-            --set infra.clusterName="${CLUSTER_NAME}" \
-            --set cloudControllerManager.imageRepository="${IMAGE_REGISTRY}" \
-            --set cloudNodeManager.imageRepository="${IMAGE_REGISTRY}" \
-            --set cloudControllerManager.imageName="${CCM_IMAGE_NAME}" \
-            --set cloudNodeManager.imageName="${CNM_IMAGE_NAME}" \
-            --set-string cloudControllerManager.imageTag="${IMAGE_TAG}" \
-            --set-string cloudNodeManager.imageTag="${IMAGE_TAG}" \
-            --set cloudControllerManager.replicas="${CCM_COUNT}" \
-            --set cloudControllerManager.enableDynamicReloading="${ENABLE_DYNAMIC_RELOADING}"  \
-            --set cloudControllerManager.cloudConfig="${CLOUD_CONFIG}" \
-            --set cloudControllerManager.cloudConfigSecretName="${CONFIG_SECRET_NAME}" \
-            --set cloudControllerManager.logVerbosity="${CCM_LOG_VERBOSITY}" \
-            --set-string cloudControllerManager.clusterCIDR="${CCM_CLUSTER_CIDR}"
-    fi
 
-    export -f wait_for_nodes
-    timeout --foreground 1800 bash -c wait_for_nodes
+        export -f wait_for_nodes
+        timeout --foreground 1800 bash -c wait_for_nodes || exit 1
```

Comment:
do we need the explicit exit? won't the command already exit non-zero if it fails?

Reply:
ACK